src/perception/test/ransac_test.cpp

```diff
@@ -0,0 +1,183 @@
+
+    ASSERT_EQ(ground_removed_cloud.points.size(), 2);
+}
+
+/**
+ * @brief Test Scenario: The points in the plane and a close enough point is removed - 1 point left.
+ * 
+ */
+TEST_F(RANSACTest, Test3) {
+    auto ground_removal = new RANSAC(0.5, 100);
+
+    pcl::PointCloud<pcl::PointXYZI> ground_removed_cloud;
+    ground_removal->groundRemoval(pcl_cloud, ground_removed_cloud);
+
+    ASSERT_EQ(ground_removed_cloud.points.size(), 1);
+}
+
+
+/**
+ * @brief Test Scenario: The epsilon threshold is set to 0 - No points are removed.
+ * 
+ */
+TEST_F(RANSACTest, Test4) {
+    auto ground_removal = new RANSAC(0, 10);
+
+    pcl::PointCloud<pcl::PointXYZI> ground_removed_cloud;
+    ground_removal->groundRemoval(pcl_cloud, ground_removed_cloud);
+
+    ASSERT_EQ(ground_removed_cloud.points.size(), 5);
+}
+
+
+/**
+ * @brief Test Scenario: Number of repetitions is set to 0 - Expected a point cloud with 0 points.
+ * 
+ */
+TEST_F(RANSACTest, Test5) {
+    auto ground_removal = new RANSAC(100, 0);
+
+    pcl::PointCloud<pcl::PointXYZI> ground_removed_cloud;
+    ground_removal->groundRemoval(pcl_cloud, ground_removed_cloud);
+
+    ASSERT_EQ(ground_removed_cloud.points.size(), 0);
+}
+
+
+/**
+ * @brief Test Scenario: Really small threshold. Only the points of the plane are considered as part of the ground.
+ * 
+ */
+TEST_F(RANSACTest, Test6) {
+    auto ground_removal = new RANSAC(0.00000000000000001, 10);
+
+    pcl::PointCloud<pcl::PointXYZI> ground_removed_cloud;
+    ground_removal->groundRemoval(pcl_cloud, ground_removed_cloud);
+
+    ASSERT_EQ(ground_removed_cloud.points.size(), 2);
+}
+
+
+/**
+ * @brief Test Scenario: Really great threshold - All points are considered as part of the plane.
+ * 
+ */
+TEST_F(RANSACTest, Test7) {
+    auto ground_removal = new RANSAC(1000000, 1);
+
+    pcl::PointCloud<pcl::PointXYZI> ground_removed_cloud;
+    ground_removal->groundRemoval(pcl_cloud, ground_removed_cloud);
+
+    ASSERT_EQ(ground_removed_cloud.points.size(), 0);
+}
+
+
+/**
+ * @brief Test Scenario: Point cloud with only 3 points: 0 points are expected after the ground removal.
+ * 
+ */
+TEST_F(RANSACTest, Test8) {
+    auto ground_removal = new RANSAC(100, 100);
+
+    pcl::PointCloud<pcl::PointXYZI> ground_removed_cloud;
+    ground_removal->groundRemoval(pcl_cloud_3_points, ground_removed_cloud);
+
+    ASSERT_EQ(ground_removed_cloud.points.size(), 0);
+}
+
+/**
+ * @brief Test Scenario: Point cloud with only 3 points: The epsilon threshold is set to 0 - No points are removed.
+ * 
+ */
+TEST_F(RANSACTest, Test9) {
+    auto ground_removal = new RANSAC(0, 100);
+
+    pcl::PointCloud<pcl::PointXYZI> ground_removed_cloud;
+    ground_removal->groundRemoval(pcl_cloud_3_points, ground_removed_cloud);
+
+    ASSERT_EQ(ground_removed_cloud.points.size(), 3);
+}
+
+/**
+ * @brief Test Scenario: Point cloud with 0 points - Must return a point with 0 points also.
+ * 
+ */
+TEST_F(RANSACTest, Test10) {
```

Comment:
Tests are enough to check functionality, I'll just leave the suggestion to change the name of the tests to something more alusive of the tests itself, but as it is already in the comments, I leave it to you